stream_alert/alert_processor/outputs/pagerduty.py

```diff
@@ -479,8 +592,38 @@ def dispatch(self, **kwargs):
         incidents_url = self._get_endpoint(self._base_url, self.INCIDENTS_ENDPOINT)
 
         try:
-            success = self._post_request_retry(incidents_url, incident, self._headers, True)
+            incident = self._post_request_retry(incidents_url, incident_data, self._headers, True)
         except OutputRequestFailure:
-            success = False
+            incident = False
 
-        return self._log_status(success)
+        if not incident:
+            LOGGER.error('Could not create main incident, %s', self.__service__)
+            return self._log_status(False)
+
+        # Extract the incident id from the incident that was just created
+        incident_id = incident.get('incident').get('id')
+
+        # Create alert to hold all the incident details
+        event_data = events_v2_data(creds['integration_key'], **kwargs)
+        event = self._create_event(event_data)
+        if not event:
+            LOGGER.error('Could not create incident event, %s', self.__service__)
+            return self._log_status(False)
+
+        # Lookup the incident_key returned as dedup_key to get the incident id
+        incident_key = event.get('dedup_key')
```

Comment:
What would occur here if this `.get('dedup_key')` returns `None`? Should it be evaluated before continuing on?

Reply:
This is a good point, since the API will return all open incidents if the `incident_key` is empty, and that can cause bad things because it will pick up the first element of the results to be merged.

Reply:
@javuto good to know! any ideas on how we can account for that?

Reply:
I just pushed some code that fails the incident creation and logs it, that should be easy to spot and does not put any alerts/incidents at risk to be merged with the wrong thing.